src/System.Drawing.Common/tests/Printing/PrintDocumentTests.cs

```diff
@@ -0,0 +1,272 @@
+// See the LICENSE file in the project root for more information.
+//
+// Copyright (C) 2005-2006 Novell, Inc (http://www.novell.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using Xunit;
+
+namespace System.Drawing.Printing.Tests
+{
+    public class PrintDocumentTests
+    {
+        private readonly PageSettings _pageSettings = new PageSettings()
+        {
+            PaperSize = new PaperSize()
+            {
+                RawKind = (int)PaperKind.A3
+            }
+        };
+
+        [Fact]
+        public void Ctor_Default_Success()
+        {
+            using (var document = new PrintDocument())
+            {
+                Assert.Equal("document", document.DocumentName);
+                Assert.False(document.OriginAtMargins);
+                AssertDefaultPageSettings(document.DefaultPageSettings);
+            }
+        }
+
+        [Fact]
+        public void DefaultPageSettings_SetValue_ReturnsExpected()
+        {
+            using (var document = new PrintDocument())
+            {
+                document.DefaultPageSettings = null;
+                Assert.IsAssignableFrom<PageSettings>(document.DefaultPageSettings);
+
+                document.DefaultPageSettings = _pageSettings;
+                Assert.Equal(_pageSettings.PaperSize.Kind, _pageSettings.PaperSize.Kind);
+            }
+        }
+
+        [Fact]
+        public void DefaultPageSettings_Null_ReturnsExpected()
+        {
+            using (var document = new PrintDocument())
+            {
+                document.DefaultPageSettings = null;
+                AssertDefaultPageSettings(document.DefaultPageSettings);
+            }
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("newDocument")]
+        public void DocumentName_SetValue_ReturnsExpected(string documentName)
+        {
+            if (documentName == null)
```

Comment:
Oh I was thinking about something else when I wrote this lol. I probably should write the null case in separate test to assert that if document name is set to null it will return empty string ...